cpp/foxglove/include/foxglove/error.hpp

```diff
@@ -0,0 +1,38 @@
+
+#include <exception>
+#include <memory>
+
+struct foxglove_error;
+
+namespace foxglove {
+
+enum class FoxgloveErrorKind {
+  Unspecified,
+  ValueError,
+  SinkClosed,
+  SchemaRequired,
+  MessageEncodingRequired,
+  ServerAlreadyStarted,
+  Bind,
+  DuplicateChannel,
+  DuplicateService,
+  MissingRequestEncoding,
+  ServicesNotSupported,
+  ConnectionGraphNotSupported,
+  IoError,
+  McapError,
+};
+
+class FoxgloveError : public std::exception {
+public:
+  FoxgloveError(const foxglove_error&& error);
```

Comment:
also not a big deal but another surprise-avoiding habit (only matters for constructors with 1 argument iirc)
```suggestion
  explicit FoxgloveError(const foxglove_error&& error);
```

Reply:
Separately, this line caught my eye as `const foo&&` is not commonly used.....but reading more about it, I think you could argue that it's correct/defensible to use here https://www.sandordargo.com/blog/2021/08/18/const-rvalue-references

Reply:
It was muscle memory for me, maybe incorrectly. I'll have to read the blog post and think it over